src/overlays/actors/ovl_kaleido_scope/z_kaleido_equipment.c

```diff
@@ -1,6 +1,595 @@
+    textureHeight = 4096 / (width * 2);
+    textureSize = width * textureHeight * 2;
+    textureCount = remainingSize / textureSize;
+    if ((remainingSize % textureSize) != 0) {
+        textureCount += 1;
+    }
+
+    vtxIndex = 80;
+
+    gSetTileCustom(POLY_OPA_DISP++, G_IM_FMT_RGBA, G_IM_SIZ_16b, width, textureHeight, 0, G_TX_NOMIRROR | G_TX_CLAMP,
+                   G_TX_NOMIRROR | G_TX_CLAMP, G_TX_NOMASK, G_TX_NOMASK, G_TX_NOLOD, G_TX_NOLOD);
+
+    remainingSize -= textureSize;
+
+    for (i = 0; i < textureCount; i++) {
+        gSPVertex(POLY_OPA_DISP++, &pauseCtx->equipVtx[vtxIndex], 4, 0);
+
+        gDPSetTextureImage(POLY_OPA_DISP++, G_IM_FMT_RGBA, G_IM_SIZ_16b, width, curTexture);
+
+        gDPLoadSync(POLY_OPA_DISP++);
+        gDPLoadTile(POLY_OPA_DISP++, G_TX_LOADTILE, 0, 0, (width - 1) << 2, (textureHeight - 1) << 2);
+
+        gSP1Quadrangle(POLY_OPA_DISP++, 0, 2, 3, 1, 0);
+
+        curTexture += textureSize;
+
+        if ((remainingSize - textureSize) < 0) {
+            if (remainingSize > 0) {
+                textureHeight = remainingSize / (s32)(width * 2);
+                remainingSize -= textureSize;
+
+                gSetTileCustom(POLY_OPA_DISP++, G_IM_FMT_RGBA, G_IM_SIZ_16b, width, textureHeight, 0,
+                               G_TX_NOMIRROR | G_TX_CLAMP, G_TX_NOMIRROR | G_TX_CLAMP, G_TX_NOMASK, G_TX_NOMASK,
+                               G_TX_NOLOD, G_TX_NOLOD);
+            }
+        } else {
+            remainingSize -= textureSize;
+        }
+
+        vtxIndex += 4;
+    }
+
+    CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_kaleido_equipment.c", 122);
+}
+
+void KaleidoScope_DrawPlayerWork(GlobalContext* globalCtx) {
+    PauseContext* pauseCtx = &globalCtx->pauseCtx;
+    Vec3f pos;
+    Vec3s rot;
+    f32 scale;
+
+    if (LINK_AGE_IN_YEARS == YEARS_CHILD) {
+        pos.x = 2.0f;
+        pos.y = -130.0f;
+        pos.z = -150.0f;
+        scale = 0.046f;
+    } else if (CUR_EQUIP_VALUE(EQUIP_SWORD) != 2) {
+        pos.x = 25.0f;
+        pos.y = -228.0f;
+        pos.z = 60.0f;
+        scale = 0.056f;
+    } else {
+        pos.x = 20.0f;
+        pos.y = -180.0f;
+        pos.z = -40.0f;
+        scale = 0.047f;
+    }
+
+    rot.y = 32300;
+    rot.x = rot.z = 0;
+    func_8009214C(globalCtx, pauseCtx->playerSegment, &pauseCtx->playerSkelAnime, &pos, &rot, scale,
+                  CUR_EQUIP_VALUE(EQUIP_SWORD), CUR_EQUIP_VALUE(EQUIP_TUNIC) - 1, CUR_EQUIP_VALUE(EQUIP_SHIELD),
+                  CUR_EQUIP_VALUE(EQUIP_BOOTS) - 1);
+}
+
+void KaleidoScope_DrawEquipment(GlobalContext* globalCtx) {
+    PauseContext* pauseCtx = &globalCtx->pauseCtx;
+    Input* input = &globalCtx->state.input[0];
+    u16 i;
+    u16 j;
+    u16 k;
+    u16 bit;
+    u16 temp;
+    u16 point;
+    u16 rowStart;
+    u16 pad;
+    s16 cursorMoveResult;
+    u16 cursorItem;
+    u16 cursorSlot;
+    s16 cursorPoint;
+    s16 cursorX;
+    s16 cursorY;
+    volatile s16 oldCursorPoint;
```

Comment:
Is this really needed?

Reply:
It's clearly fake but yeah unfortunately, this function needed like 5 different fake match things in order to match and this is the only solution I could find. It was a real pain.